neqo-http3/src/features/extended_connect/webtransport_session.rs

```diff
@@ -477,7 +478,7 @@ impl Session {
         Ok(())
     }
 
-    pub fn datagram(&self, datagram: Vec<u8>) {
+    pub fn datagram(&self, datagram: VecDeque<u8>) {
```

Comment:
https://github.com/mozilla/neqo/pull/2796/ will improve some of this by passing a `datagram: &[u8]` instead of a `datagram: Vec<u8>`, thus not needing a `Vec` allocation on the caller side.

See `session.rs` here:

https://github.com/mozilla/neqo/pull/2796/files#diff-01aa8f1aba4108ba00b94f619d3ee9de48583de59a33f05e526e480b1e8acf59R388-R402